src/qt/clientmodel.cpp

```diff
@@ -30,14 +31,23 @@ ClientModel::ClientModel(interfaces::Node& node, OptionsModel *_optionsModel, QO
     optionsModel(_optionsModel),
     peerTableModel(nullptr),
     banTableModel(nullptr),
-    pollTimer(nullptr)
+    m_thread(new QThread(this))
 {
     cachedBestHeaderHeight = -1;
     cachedBestHeaderTime = -1;
     peerTableModel = new PeerTableModel(m_node, this);
     banTableModel = new BanTableModel(m_node, this);
-    pollTimer = new QTimer(this);
-    connect(pollTimer, &QTimer::timeout, this, &ClientModel::updateTimer);
+
+    pollTimer = new QTimer;
+    // move timer to thread so that polling doesn't disturb event loop
+    pollTimer->moveToThread(m_thread);
+    m_thread->start();
+    connect(pollTimer, &QTimer::timeout, [this] {
```

Comment:
Can't you leave `ClientModel::updateTimer` as-is for now? Or at least move the code movement to another commit?

Reply:
Yes but I'd have to connect with Qt::Direct connection, otherwise the slot would be called in the GUI event loop.

Reply:
Even after `moveToThread`? Because of the slot being on a GUI object?

Reply:
Yes, with auto connection (the default) the signal checks if the target's thread is the same as the sender or not, if not then it's the same as queued connection.

Reply:
I think you should do this connect before moving the timer to the thread?

Reply:
`QObject::connect` is thread safe.